pagebench: add `basebackup` gRPC support

erikgrinaker opened the pull request:
## Problem

Pagebench does not support gRPC for `basebackup` benchmarks.

Requires #12243.
Touches #11728.

## Summary of changes

Add gRPC support via gRPC connstrings, e.g. `pagebench basebackup --page-service-connstring grpc://localhost:51051`.

Also change `--gzip-probability` to `--no-compression`, since this must be specified per-client for gRPC.

hlinnaka commented on pageserver/pagebench/src/cmd/basebackup.rs:
```diff
@@ -270,3 +268,90 @@ async fn client(
     all_work_done_barrier.wait().await;
 }
+
+/// A basebackup client. This allows swithcing out the client protocol implementation.
```
```suggestion
/// A basebackup client. This allows switching out the client protocol implementation.
```

hlinnaka commented on pageserver/pagebench/src/cmd/basebackup.rs:
```diff
@@ -270,3 +268,90 @@ async fn client(
+        lsn: Option<Lsn>,
+        gzip: bool,
+    ) -> anyhow::Result<Pin<Box<dyn AsyncRead + Send + 'static>>> {
+        // TODO: support gzip compression.
+        ensure!(!gzip, "gRPC client does not support gzip compression yet");
```
what would it take to support gzip?

erikgrinaker replied:
Not much, probably an hour or two of work. I plan to get to it in the next couple of days, it's blocked on some other pending PRs.

erikgrinaker commented:
I'm all compression to this once #12280 merges.

erikgrinaker commented:
Added compression support here and rewrote it to use `page_api::Client`. Changed `--gzip-probability` to `--no-compression`. I'll take the approvals to still be valid, TFTRs!
